core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java

```diff
@@ -937,6 +948,14 @@ public NativeImageInvokerInfo build() {
                     // --enable-monitoring=heapdump is not supported on Windows
                     monitoringOptions.add(NativeConfig.MonitoringOption.HEAPDUMP);
                 }
+
+                if (nativeMonitoringItems != null && !nativeMonitoringItems.isEmpty()) {
+                    monitoringOptions.addAll(nativeMonitoringItems.stream()
+                            .map(NativeMonitoringBuildItem::getOption)
+                            .distinct()
```

Comment:
I wonder if we should switch `monitoringOptions` to a `LinkedHashSet` instead of a `List`.

@roberttoyonaga WDYT?